### DISCUSSION TITLE
questions about create_index

### DISCUSSION THREAD

**Discussion by winterfellding** [milvus-io/milvus#4841] [Q&A and General discussion] [answered]
Hi, I have 80 million 256 dimension image vector to insert into milvus server, after I create_collection, insert data，create_index, I've got some questions
1. the create_index is very slow, I ran on aws i3.2x machine, took about 5 days to run the create_index phase, is that because the kNN or kmeans clustering algorithms runs slow on big dataset?
2. I have some customize proerty to insert for each vector data, I see this is an experimental feature on the github homepage, how can I do it with the insert API
3. if I have some new image vector data to insert, how can I do the incremental update index, should I just insert the new data to the collection and rerun the create_index API, is it fully reconstruct the index or just incremental update the index file

**Comment by shengjun1985** [comment, marked as answer]
> took about 5 days to run the create_index phase

Would you like to provide more information such as Milvus version, index file size, index_type and index parameters?
The time of building IVF index is greatly affected by the parameter 'nlist'.

**Reply by winterfellding** [reply]
Milvus version: v0.10.0
index_file_size: default, 1024 I think
index_type: IVF_SQ8
nlist: 32768

**Reply by shengjun1985** [reply]
From your parameters, the row count of each segment may be about 1,000,000.
So the 'nlist' 32768 seems too large. You can try to use 2048 or 4096.
